src/mod_ping.erl

```diff
@@ -133,7 +136,11 @@ maybe_add_hooks_handlers(Host, true) ->
     ejabberd_hooks:add(user_send_packet, Host,
                        ?MODULE, user_send, 100),
     ejabberd_hooks:add(user_sent_keep_alive, Host,
-                       ?MODULE, user_keep_alive, 100);
+                       ?MODULE, user_keep_alive, 100),
+    ejabberd_hooks:add(user_ping_timeout, Host,
```

Comment:
I don't see these new hook handlers removed in the `stop/1` function. I recommend to extract definition of all the hook handlers in the module to a function and use:
* function `ejabberd_hooks:add/1` in `mod_ping:start/2`
* function `ejabberd_hooks:delete/1` in `mod_ping:stop/1`

As an example you can take `mod_roster`.

Reply:
I overlooked deletion. Will fix it shortly.